mirrord-layer/src/file/hooks.rs

```diff
@@ -132,50 +135,37 @@ pub(super) unsafe extern "C" fn openat_detour(
     raw_path: *const c_char,
     open_flags: c_int,
 ) -> RawFd {
-    let path: PathBuf = match CStr::from_ptr(raw_path).to_str().map_err(|fail| {
-        error!(
-            "Failed converting raw_path {:#?} from `c_char` with {:#?}",
-            raw_path, fail
-        );
-        -1
-    }) {
-        Ok(path_str) => path_str.into(),
-        Err(fail) => return fail,
-    };
-
-    // `openat` behaves the same as `open` when the path is absolute.
-    if path.is_absolute() {
-        open_detour(raw_path, open_flags)
-    } else if fd == AT_FDCWD {
-        error!(
-            r"Failed `openat_detour` as it cannot deal with `AT_FDCWD`,
-            cannot handle cwd in remote context.
-            `openat_detour` was called with arguments:
-            fd {:#?} | raw_path {:#?} | open_flags {:#?}",
-            fd, raw_path, open_flags
-        );
+    let remote_fd = OPEN_FILES.lock().unwrap().get(&fd).cloned();
+    if let Some(remote_fd) = remote_fd {
+        let path: PathBuf = match CStr::from_ptr(raw_path).to_str().map_err(|fail| {
+            error!(
+                "Failed converting raw_path {:#?} from `c_char` with {:#?}",
+                raw_path, fail
+            );
+            -1
+        }) {
+            Ok(path_str) => path_str.into(),
+            Err(fail) => return fail,
+        };
 
-        -1
-    } else {
-        // Relative path requires special handling, we must identify the relative part (relative to
-        // what).
-        let remote_fd = OPEN_FILES.lock().unwrap().get(&fd).cloned();
+        // Todo(check): AT_FDCWD - refers to the CWD of the calling process.
+        // Do we really need to manipulate paths here since we already have the path for the given
+        // fd in the agent? since AT_FDCWD refers to the context of the calling process, do
+        // we need to check if the remote fd even exists?
 
-        // Are we managing the relative part?
-        if let Some(remote_fd) = remote_fd {
+        if path.is_absolute() || fd == AT_FDCWD {
```

Comment:
The docs on this handling are a bit confusing to me, but shouldn't these be handled differently?

> absolute path, it ignores the fd argument

> AT_FDCWD in the fd parameter, the current working directory is used